project/interactivemap/settings.py

```diff
@@ -162,10 +162,10 @@
     'interactivemap-frontend-*.now.sh',
 )
 
-AUTH0_DOMAIN = os.environ.get('AUTH0_DOMAIN', 'chronoscio.auth0.com')
```

Comment:
Removed this default because it doe snot seem to work when combined with empty strings.

Reply:
I don't know what you mean

Reply:
```
Traceback (most recent call last):
  File "project/manage.py", line 29, in <module>
    execute_from_command_line(sys.argv)
  File "/home/tokumaru/.local/lib/python3.5/site-packages/django/core/management/__init__.py", line 371, in execute_from_command_line
    utility.execute()
  File "/home/tokumaru/.local/lib/python3.5/site-packages/django/core/management/__init__.py", line 317, in execute
    settings.INSTALLED_APPS
  File "/home/tokumaru/.local/lib/python3.5/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/home/tokumaru/.local/lib/python3.5/site-packages/django/conf/__init__.py", line 43, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/tokumaru/.local/lib/python3.5/site-packages/django/conf/__init__.py", line 106, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/home/tokumaru/chronoscio/backend/project/interactivemap/settings.py", line 174, in <module>
    jwks = json.loads(jsonurl.read())
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```
I traced it back to precisely this one-line change and checked several times why flipping back and forth between line options. It happens if `AUTH0_DOMAIN` is `'chronoscio.auth0.com'` while the other two are empty strings or None. However, if all 3 are None, it works fine. Might only be a problem when hosting locally.

Reply:
What are the exact values of your environment variables?

Reply:
The edge case I am describing is when you have none of the django.env sourced. When that happens it can still work with the defaults specified in settings.py except for this. When all the env vars are set, there is no need for defaults. The problem I observed is that the url is not compatible with empty strings or Nones, but all Nones are compatible.